06-building-a-program-start-to-finish/writing-tests/output/spec.js

```diff
@@ -0,0 +1,142 @@
+
+  it('Should return true for creating Index', () => {
+    expect(invertedIndex.createIndex('books.json', myBook)).toBeTruthy();
+  });
+
+  it('Should return `No file has been indexed yet', () => {
+    const search = () => {
+      invertedIndex.searchIndex('alice');
+    };
+    expect(search).toThrowError();
+  });
+
+  it('Should return an empty if not found', () => {
+    invertedIndex.searchIndex('alive', ['books.json']);
+    expect(invertedIndex.finalResult).toEqual({ 'books.json': Object({ alive: [] }) });
+  });
+
+  it('Should return `please enter a keyword to search.`', () => {
+    const search = () => {
+      invertedIndex.searchIndex();
+    };
+    expect(search).toThrowError('please enter a keyword to search.');
+  });
+
+  it('Should return true for well formatted File', () => {
+    const book = JSON.stringify(myBook);
+    expect(InvertedIndex.validateFile(book, 'books.json')).toBeTruthy();
+  });
+
+  it('Should throw an error for empty books', () => {
+    const fileName = 'emptybook.json';
+    const checkEmptyBook = () => {
+      InvertedIndex.validateFile(emptyBook, fileName);
+    };
+    expect(checkEmptyBook).toThrowError();
+  });
+
+  it('Should throw an error for Invalid keys', () => {
+    const name = 'invalidBook.json';
+    const invalidKey = () => {
+      InvertedIndex.validateFile(invalidKeys, name);
+    };
+    expect(invalidKey).toThrowError(`${name} is not well formatted`);
+  });
+
+  it('Should return an object for getIndex method', () => {
+    expect(typeof getIndex).toBe('object');
+  });
+
+  it('Should return an alive as the first token', () => {
+    const alltoken = Object.keys(getIndex);
+    expect(alltoken[0]).toBe('alice');
+  });
+
+  it('Should return the numbers of books in a file', () => {
+    const numOfBooks = invertedIndex.booksIndex('books.json');
+    expect(numOfBooks.length).toEqual(6);
+  });
+
+  it('Should return an array for the JSON File', (done) => {
+    const readFile = InvertedIndex.readFile(jsonFile);
+    readFile.then((res) => {
+      expect(res[1][0].title).toBe('Alice in Wonderland');
+      done();
+    });
+  });
+
+  it('Should return throw an error for Invalid file Extension', () => {
+    const validate = () => {
+      InvertedIndex.validateFile('result', 'sample.txt');
+    };
+    const error = 'sample.txt has an Invalid File extension, JSON only';
+    expect(validate).toThrowError(error);
+  });
+
+  it('Should return an array of clean values', () => {
+    expect(InvertedIndex.cleanValues('How, are, you doing today'))
+    .toEqual(['how', 'are', 'you', 'doing', 'today']);
+  });
+
+  it('Should return', () => {
+    expect(invertedIndex.getResult('alice', ['books.json']))
+    .toEqual({ alice: [0, 3] });
+  });
+
+  it('Should return', () => {
+    const validInput = () => {
+      invertedIndex.getResult('[[[[[[[[[[[]]]]]]]]]', ['books.json']);
+    };
+    expect(validInput).toThrow('Search for Aplhanumeric values only');
+  });
+});
```

Comment:
Please you will need to have separate Test suites for your methods.

Generally the language used on your tests should be understood by both the developer and the customer, especially to the level of a customer's understanding.

**### Sample**
```
describe("Calculator: add", () => {

    it("should be able to add 2 numbers together", () => {
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", () => {
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```
The result of this will read:
Calculator: add
    _should be able to add 2 numbers together_
   _should be able to add 3 numbers together_

Please review the language used in your test. Paying more attention to the language used on your "it" and 'describe' functions